docs/sources/reference/builder.md

```diff
@@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
 
 The `ENV` instruction sets the environment variable `<key>` to the value
 `<value>`. This value will be passed to all future 
-`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
-functionally equivalent to prefixing the command with `<key>=<value>`
+`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
+command without affecting future commands or containers, prefix the command
+with `<key>=<value>`.
```

Comment:
is that true?  In one window I did:
docker run -ti -e foo=bar ubuntu bash
and in another I did:
docker exec -ti ce0fd7706d0a bash
ce0f.. is the 1st container ID and I see foo set to bar. 
This also works if I use ENV from the Dockerfile used to create the first container/image and then left off the -e on the run.

Personally, I wouldn't talk about "RUN foo=bar do-something...".  That's just normal shell processing.

How about:
The `ENV` instruction sets the environment variable `<key>` to the value `<value>`.  This variable will not only be visible within the container but also available for environment variable expansion within the following Dockerfile commands: ...